src/components/PostDetails.tsx

```diff
@@ -3,59 +3,35 @@ import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
-
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addNewComment,
+  changeError,
+  fetchComments,
+  removeComment,
+} from '../features/commentsSlice';
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-  const [visible, setVisible] = useState(false);
-
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  const dispatch = useAppDispatch();
+  const {
+    comments,
+    loading: loaded,
```

Comment:
The variable name 'loaded' might be misleading. It seems to indicate whether the comments are loaded, but it's used as a boolean flag for loading state. Consider renaming it to 'isLoading' or 'loading' for clarity.